src/main/java/seedu/contax/logic/commands/BatchCommand.java

```diff
@@ -85,11 +85,11 @@ public CommandResult execute(Model model) throws CommandException {
         }
         List<CommandResult> commandResultList = new ArrayList<>();
         Person restorePerson = model.getFilteredPersonList().get(indexList.get(0).getZeroBased());
-        for (Index index: indexList) {
+        for (int i = indexList.get(indexList.size() - 1).getOneBased(); i >= indexList.get(0).getOneBased(); i--) {
```

Comment:
This loop does not look right, it is running on continuous indexes (which may not be the case in batch).

Reply:
Ohh my careless, thx changed